contrib/99designs/gqlgen/tracer.go

```diff
@@ -0,0 +1,140 @@
+// Options can be passed in for further configuration.
+func New(opts ...Option) graphql.Tracer {
+	var t gqlTracer
+	defaults(&t.cfg)
+	for _, o := range opts {
+		o(&t.cfg)
+	}
+	return &t
+}
+
+// gqlTracer implements the graphql.Tracer interface.
+func (t *gqlTracer) StartOperationParsing(ctx context.Context) context.Context {
+	// not implemented
+	return ctx
+}
+
+// gqlTracer implements the graphql.Tracer interface.
+func (t *gqlTracer) EndOperationParsing(ctx context.Context) {
+	// not implemented
+}
+
+// gqlTracer implements the graphql.Tracer interface.
+func (t *gqlTracer) StartOperationValidation(ctx context.Context) context.Context {
+	// not implemented
+	return ctx
+}
+
+// gqlTracer implements the graphql.Tracer interface.
+func (t *gqlTracer) EndOperationValidation(ctx context.Context) {
+	// not implemented
+}
+
+func (t *gqlTracer) StartOperationExecution(ctx context.Context) context.Context {
+	rctx := graphql.GetRequestContext(ctx)
+	name := defaultResourceName
+	if rctx != nil && rctx.OperationName != "" {
+		name = rctx.OperationName
+	}
+	opts := []ddtrace.StartSpanOption{
+		tracer.SpanType(ext.SpanTypeGraphql),
+		tracer.ResourceName(name),
+		tracer.ServiceName(t.cfg.serviceName),
+	}
+	if !math.IsNaN(t.cfg.analyticsRate) {
+		opts = append(opts, tracer.Tag(ext.EventSampleRate, t.cfg.analyticsRate))
+	}
+	if s, ok := tracer.SpanFromContext(ctx); ok {
+		opts = append(opts, tracer.ChildOf(s.Context()))
+	}
+	_, ctx = tracer.StartSpanFromContext(ctx, name, opts...)
+	return ctx
+}
+
+func (t *gqlTracer) StartFieldExecution(ctx context.Context, field graphql.CollectedField) context.Context {
+	span, ctx := tracer.StartSpanFromContext(ctx, "Field_"+field.Name)
+	span.SetTag("field", field.Name)
+	return ctx
+}
+
+func (t *gqlTracer) StartFieldResolverExecution(ctx context.Context, rc *graphql.ResolverContext) context.Context {
+	// This is the span created in StartFieldExecution.
+	// StartFieldResolverExecution is called only once per StartFieldExecution, so we can add context to the
+	// span rather than starting a child span.
+	span, ok := tracer.SpanFromContext(ctx)
+	if !ok {
+		return ctx
+	}
+	span.SetTag(ext.SpanName, rc.Object+"_"+rc.Field.Name)
+	span.SetTag(tagResolverObject, rc.Object)
+	span.SetTag(tagResolverField, rc.Field.Name)
+	return ctx
+}
+
+// gqlTracer implements the graphql.Tracer interface.
+func (t *gqlTracer) StartFieldChildExecution(ctx context.Context) context.Context {
+	// not implemented
+	return ctx
+}
+
+func (t *gqlTracer) EndFieldExecution(ctx context.Context) {
+	span, ok := tracer.SpanFromContext(ctx)
+	if !ok {
+		return
+	}
+	defer span.Finish()
+	resCtx := graphql.GetResolverContext(ctx)
+	reqCtx := graphql.GetRequestContext(ctx)
+	if resCtx == nil || reqCtx == nil {
+		return
+	}
+	errList := reqCtx.GetErrors(resCtx)
+	if len(errList) != 0 {
+		span.SetTag(ext.Error, true)
+		for idx, err := range errList {
+			span.SetTag(fmt.Sprintf("gqlgen.error_%d.message", idx), err.Error())
```

Comment:
I'm not sure about this `gqlgen.*` prefix. It creates a collapsible group in the UI and it may make it look like everything outside of it is _not_ `gqlgen` related. Can we either remove it or have it as `gqlgen_error`. Would that be better?